proxy/gateway/nodejs/test/test_endpoint.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+let nano = require('nanomsg');
+let unlink = require('fs').unlink;
+
+class TestEndpoint {
+  constructor(id) {
+    this.id = id;
+    this.socket = nano.socket('pair');
+    this.socket.connect('ipc://' + id);
+    this._received = new Promise((resolve) => {
+      this.socket.on('data', data => resolve(data));
+    });
+  }
+
+  send(msg) { this.socket.send(msg); }
+
+  receive() { return this._received; }
```

Comment:
Yes, that's how I'm using it. But this is a test class, only used in tests to act as the _other_ side of a nanomsg session. Because I implement the receive function as a Promise, it allows me to do stuff like this:
```
    return listener.receive()
      .should.eventually.deep.equal(Buffer.from('hello'));
```